lib/helpers/node-matcher.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+// Source: jscodeshift/src/matchNode.js
+const hasOwn = Object.prototype.hasOwnProperty.call.bind(Object.prototype.hasOwnProperty);
+
+function isNode(value) {
+  return typeof value === 'object' && value;
+}
+
+function matchNode(haystack, needle) {
+  if (typeof needle === 'function') {
+    return needle(haystack);
+  }
+  if (isNode(needle) && isNode(haystack)) {
+    return Object.keys(needle).every(function (property) {
+      return hasOwn(haystack, property) && matchNode(haystack[property], needle[property]);
+    });
+  }
+  return haystack === needle;
+}
+
+// Source: jscodeshift/src/core.js
+function matchOne(path, filter) {
+  if (typeof path.get === 'function') {
+    path = path.get();
+  } else {
+    path = { value: path };
+  }
+  return matchNode(path.value, filter);
+}
+
+// Wrappers
+function nodeMatchesRef(testNode, refNode) {
+  return testNode && refNode && matchOne(testNode, refNode);
+}
```

Comment:
p3/5

I think we are guaranteed that `testNode`/`refNode` are not falsey, right? Should we drop this func and use `matchOne` throughout?

Reply:
So long as the plan is to have a single-export strategy (currently `match`) that does this verification immediately, I see no reason not to get rid of the wrapper for `matchOne` altogether. 👍

Reply:
Update: addressed in 8023b15 👍